extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -122,82 +125,84 @@ public function signupForm() {
-
-			$this->result = $response->getVal( 'result', '' );
-			$this->msg = $response->getVal( 'msg', '' );
-			$this->errParam = $response->getVal( 'errParam', '' );
-
-			if ( $this->result == 'ok' ) {
-
-				/*
-				 * Remove when SOC-217 ABTest is finished
-				 */
-				$signupForm = new UserLoginForm( $this->wg->request );
-
-				if ( $signupForm->isAllowedRegisterUnconfirmed() ) {
-					$user = User::newFromName( $this->username );
-					// Get and clear redirect page
-					$userSignupRedirect = $user->getOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME );
-					$user->setOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME, null );
-
-					$user->saveSettings();
-
-					// redirect user
-					if ( !empty( $userSignupRedirect ) ) {
-						// Redirect user to the point where he finished (when signup on create wiki)
-						$title = SpecialPage::getTitleFor( 'CreateNewWiki' );
-						$query = $userSignupRedirect;
-					} else {
-						$title = $user->getUserPage();
-						$query = '';
-					}
-
-					$redirectUrl = $title->getFullURL( $query );
+	public function handleSignupFormSubmit() {
+		$response = $this->app->sendRequest( 'UserSignupSpecial', 'signup' );
+		$result = $response->getVal( 'result', '' );
+
+		if ( $result == 'ok' ) {
+			/*
+			 * Remove when SOC-217 ABTest is finished
+			 */
+			$signupForm = new UserLoginForm( $this->wg->request );
+
+			if ( $signupForm->isAllowedRegisterUnconfirmed() ) {
+				$user = User::newFromName( $this->username );
+				// Get and clear redirect page
+				$userSignupRedirect = $user->getOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME );
+				$user->setOption( UserLoginSpecialController::SIGNUP_REDIRECT_OPTION_NAME, null );
+
+				$user->saveSettings();
+
+				// redirect user
+				if ( !empty( $userSignupRedirect ) ) {
+					// Redirect user to the point where he finished (when signup on create wiki)
+					$title = SpecialPage::getTitleFor( 'CreateNewWiki' );
+					$query = $userSignupRedirect;
 				} else {
+					$title = $user->getUserPage();
+					$query = '';
+				}
+
+				$redirectUrl = $title->getFullURL( $query );
+			} else {
 				/*
 				 * end remove
 				 */
-					$params = [
-						'method' => 'sendConfirmationEmail',
-						'username' => $this->username,
-						'byemail' => intval( $this->byemail ),
-					];
-					$redirectUrl = $this->wg->title->getFullUrl( $params );
-				}
-
-				$this->wg->out->redirect( $redirectUrl );
+				$params = [
+					'method' => 'sendConfirmationEmail',
+					'username' => $this->username,
+					'byemail' => intval( $this->byemail ),
+				];
+				$redirectUrl = $this->wg->title->getFullUrl( $params );
 			}
 
+			$this->track( 'signup-successful' );
+			$this->wg->out->redirect( $redirectUrl );
+		} else {
+			$this->response->setData( [
+				'result' => $result,
+				'msg' => $response->getVal( 'msg', '' ),
+				'errParam' => $response->getVal( 'errParam', '' ),
+			] );
```

Comment:
I know it is not part of the acceptance criteria, but it would be interesting to see failed attempts logged too. It may give us more insight into why signups fail.